controllers/ingrnodefwconfig_controller.go

```diff
@@ -0,0 +1,147 @@
+
+// IngrNodeFwConfigReconciler reconciles a IngrNodeFwConfig object
+type IngrNodeFwConfigReconciler struct {
+	client.Client
+	Scheme    *runtime.Scheme
+	Log       logr.Logger
+	Namespace string
+}
+
+// +kubebuilder:rbac:groups=apps,namespace=ingress-node-firewall-system,resources=daemonsets,verbs=get;list;watch;create;update;patch;delete
+
+//+kubebuilder:rbac:groups=ingress-nodefw.ingress-nodefw,resources=ingrnodefwconfigs,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=ingress-nodefw.ingress-nodefw,resources=ingrnodefwconfigs/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=ingress-nodefw.ingress-nodefw,resources=ingrnodefwconfigs/finalizers,verbs=update
+// +kubebuilder:rbac:groups=apps,resources=daemonsets,verbs=get;list;watch;create;update;patch;delete
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the IngrNodeFwConfig object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.12.1/pkg/reconcile
+func (r *IngrNodeFwConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := r.Log.WithValues("ingress node firewall config", req.NamespacedName)
+	req.Namespace = r.Namespace
+	instance := &ingressnodefwv1alpha1.IngrNodeFwConfig{}
+	err := r.Get(ctx, req.NamespacedName, instance)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return ctrl.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return ctrl.Result{}, err
+	}
+
+	if req.Name != defaultIngressNodeFirewallCrName {
+		logger.Error(err, "Invalid IngressNode firewall config resource name", "name", req.Name)
+		return ctrl.Result{}, nil // Return success to avoid requeue
+	}
+
+	if err := r.syncIngressNodeFwConfigResources(ctx, instance); err != nil {
+		return ctrl.Result{}, err
+	}
+	return ctrl.Result{}, nil
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *IngrNodeFwConfigReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&ingressnodefwv1alpha1.IngrNodeFwConfig{}).
+		Owns(&appsv1.DaemonSet{}).
+		Complete(r)
+}
+
+func (r *IngrNodeFwConfigReconciler) syncIngressNodeFwConfigResources(ctx context.Context, config *ingressnodefwv1alpha1.IngrNodeFwConfig) error {
+	logger := r.Log.WithName("syncIngressNodeFirewallConfigResources")
+	logger.Info("Start")
+	data := render.MakeRenderData()
```

Comment:
I know this is currently way more generic, but we could also just fopen the single daemon.yaml file that we're interested in, read it into memory, and then use "k8s.io/apimachinery/pkg/util/yaml" or "gopkg.in/yaml.v2" to unmarshal the single DaemonSet that we're interested in. That would cut down on code at the cost of being less generic, of course.